In [234]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder

# 5 Questions
- Predict if patient survives stroke or not.
- Predict if patient will have disability.
- What are the main factors contributing to death?
- How does our model's predicted probability of death at 14 days/6 months align with the doctors predictions?
- What are positive factors for stroke survival?

In [235]:
stroke_trials_df = pd.read_csv('Resources/V2_International_Stroke_Trials.csv', low_memory=False)
stroke_trials_df.head()

SEX  AGE RSLEEP RATRIAL RCT RVISINF RHEP24  RSBP RCONSC RDEF1  ... DPLACE  \
0   M   69      Y     NaN   Y       Y    NaN   140      D     N  ...    NaN   
1   M   76      Y     NaN   Y       N    NaN   150      F     Y  ...    NaN   
2   F   71      N     NaN   Y       N    NaN   170      F     Y  ...    NaN   
3   M   81      N     NaN   N       N    NaN   170      F     N  ...    NaN   
4   M   78      N     NaN   N       N    NaN   170      F     Y  ...    NaN   

  DDEAD DDEADC FPLACE OCCODE FDEADC CNTRYNUM   EXPDD   EXPD6  EXPD14  
0     N    0.0      E      2    NaN       27  0.6980  0.2344  0.1054  
1     N    0.0      A      2    NaN       27  0.5389  0.1555  0.0421  
2     N    0.0      A      2    NaN       27  0.5275  0.1009  0.0323  
3     N    0.0      A      4    NaN       27  0.4021  0.1147  0.0244  
4     N    0.0      E      2    NaN       27  0.5600  0.1709  0.0441  

[5 rows x 57 columns]

In [236]:
print(stroke_trials_df.columns)

Index(['SEX', 'AGE', 'RSLEEP', 'RATRIAL', 'RCT', 'RVISINF', 'RHEP24', 'RSBP',
       'RCONSC', 'RDEF1', 'RDEF2', 'RDEF3', 'RDEF4', 'RDEF5', 'RDEF6', 'RDEF7',
       'RDEF8', 'STYPE', 'RDATE', 'RXASP', 'RXHEP', 'DASP14', 'DASPLT',
       'DLH14', 'DMH14', 'DHH14', 'ONDRUG', 'DSCH', 'DIVH', 'DAP', 'DOAC',
       'DGORM', 'DSTER', 'DCAA', 'DHAEMD', 'DCAREND', 'DTHROMB', 'DMAJNCH',
       'DDIAGISC', 'DDIAGHA', 'DDIAGUN', 'DNOSTRK', 'DRSISC', 'DRSH', 'DRSUNK',
       'DPE', 'DALIVE', 'DPLACE', 'DDEAD', 'DDEADC', 'FPLACE', 'OCCODE',
       'FDEADC', 'CNTRYNUM', 'EXPDD', 'EXPD6', 'EXPD14'],
      dtype='object')


In [237]:
stroke_trials_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   19435  19435.000000  19435   18451  19435   19435  19091   
unique      2           NaN      2       2      2       2      2   
top         M           NaN      N       N      Y       N      N   
freq    10407           NaN  13750   15282  13024   13020  18655   
mean      NaN     71.715410    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.619714    NaN     NaN    NaN     NaN    NaN   
min       NaN     16.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     99.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ... DPLACE  DDEAD       DDEADC FPLACE  \
count   19435.000000  19435  19435  ...   9726  19416  2334.000000  14933   
unique           NaN      3      3  ...      6      3          NaN      6   
top              NaN      F      Y  ...      A      N          NaN      A   
freq             NaN  14921  14099  ...   7062  17376          NaN  11607   
mean      160.159197    NaN    NaN  ...    NaN    NaN     2.404456    NaN   
std        27.610382    NaN    NaN  ...    NaN    NaN     2.172007    NaN   
min        70.000000    NaN    NaN  ...    NaN    NaN     0.000000    NaN   
25%       140.000000    NaN    NaN  ...    NaN    NaN     1.000000    NaN   
50%       160.000000    NaN    NaN  ...    NaN    NaN     1.000000    NaN   
75%       180.000000    NaN    NaN  ...    NaN    NaN     4.000000    NaN   
max       295.000000    NaN    NaN  ...    NaN    NaN     8.000000    NaN   

              OCCODE       FDEADC      CNTRYNUM         EXPDD         EXPD6  \
count   19435.000000  4364.000000  19435.000000  19435.000000  19435.000000   
unique           NaN          NaN           NaN           NaN           NaN   
top              NaN          NaN           NaN           NaN           NaN   
freq             NaN          NaN           NaN           NaN           NaN   
mean        2.328891     3.441797     20.965629      0.629215      0.229676   
std         1.068117     2.382524      8.439510      0.230564      0.194553   
min         0.000000     0.000000      1.000000      0.089100      0.003500   
25%         2.000000     1.000000     14.000000      0.439100      0.084600   
50%         2.000000     4.000000     25.000000      0.637800      0.160100   
75%         3.000000     5.000000     27.000000      0.836750      0.319200   
max         9.000000     8.000000     44.000000      0.995200      0.901200   

              EXPD14  
count   19435.000000  
unique           NaN  
top              NaN  
freq             NaN  
mean        0.094167  
std         0.104473  
min         0.005000  
25%         0.028500  
50%         0.047100  
75%         0.113100  
max         0.576100  

[11 rows x 57 columns]

In [238]:
# Find the number of na values in each column
for col in stroke_trials_df.columns:
    if stroke_trials_df[col].isna().sum() > 0:
        print(f'{col}: {stroke_trials_df[col].isna().sum()}')

RATRIAL: 984
RHEP24: 344
DASP14: 22
DASPLT: 147
DLH14: 22
DMH14: 1006
DHH14: 18451
ONDRUG: 1
DSCH: 305
DIVH: 305
DAP: 18
DOAC: 18
DGORM: 23
DSTER: 28
DCAA: 29
DHAEMD: 28
DCAREND: 1005
DTHROMB: 315
DMAJNCH: 14
DDIAGISC: 23
DDIAGHA: 26
DDIAGUN: 23
DNOSTRK: 26
DRSISC: 18
DRSH: 15
DRSUNK: 1007
DPE: 14
DALIVE: 28
DPLACE: 9709
DDEAD: 19
DDEADC: 17101
FPLACE: 4502
FDEADC: 15071


In [239]:
# Fill NaN cells with '0' in columns where 0=Unknown, and fill NaN cells with 'U' in columns where U=Unknown.
stroke_trials_df['DDEADC'] = stroke_trials_df['DDEADC'].fillna(0)
stroke_trials_df['DASP14'] = stroke_trials_df['DASP14'].fillna('U')
stroke_trials_df['DASPLT'] = stroke_trials_df['DASPLT'].fillna('U')
stroke_trials_df['DRSUNK'] = stroke_trials_df['DRSUNK'].fillna('U')
stroke_trials_df['DPLACE'] = stroke_trials_df['DPLACE'].fillna('U')
stroke_trials_df['FPLACE'] = stroke_trials_df['FPLACE'].fillna('U')
stroke_trials_df['DMH14'] = stroke_trials_df['DMH14'].fillna('U')
stroke_trials_df['DHH14'] = stroke_trials_df['DHH14'].fillna('U')
stroke_trials_df['DLH14'] = stroke_trials_df['DLH14'].fillna('U')
stroke_trials_df['DSCH'] = stroke_trials_df['DSCH'].fillna('U')
stroke_trials_df['DIVH'] = stroke_trials_df['DIVH'].fillna('U')
stroke_trials_df['DCAREND'] = stroke_trials_df['DCAREND'].fillna('U')
stroke_trials_df['DTHROMB'] = stroke_trials_df['DTHROMB'].fillna('U')
stroke_trials_df['DALIVE'] = stroke_trials_df['DALIVE'].fillna('U')
stroke_trials_df['DAP'] = stroke_trials_df['DAP'].fillna('U')
stroke_trials_df['FDEADC'] = stroke_trials_df['FDEADC'].fillna(0)
stroke_trials_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   19435  19435.000000  19435   18451  19435   19435  19091   
unique      2           NaN      2       2      2       2      2   
top         M           NaN      N       N      Y       N      N   
freq    10407           NaN  13750   15282  13024   13020  18655   
mean      NaN     71.715410    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.619714    NaN     NaN    NaN     NaN    NaN   
min       NaN     16.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     99.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ... DPLACE  DDEAD        DDEADC FPLACE  \
count   19435.000000  19435  19435  ...  19435  19416  19435.000000  19435   
unique           NaN      3      3  ...      6      3           NaN      6   
top              NaN      F      Y  ...      U      N           NaN      A   
freq             NaN  14921  14099  ...   9729  17376           NaN  11607   
mean      160.159197    NaN    NaN  ...    NaN    NaN      0.288757    NaN   
std        27.610382    NaN    NaN  ...    NaN    NaN      1.085031    NaN   
min        70.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
25%       140.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
50%       160.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
75%       180.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
max       295.000000    NaN    NaN  ...    NaN    NaN      8.000000    NaN   

              OCCODE        FDEADC      CNTRYNUM         EXPDD         EXPD6  \
count   19435.000000  19435.000000  19435.000000  19435.000000  19435.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        2.328891      0.772833     20.965629      0.629215      0.229676   
std         1.068117      1.826785      8.439510      0.230564      0.194553   
min         0.000000      0.000000      1.000000      0.089100      0.003500   
25%         2.000000      0.000000     14.000000      0.439100      0.084600   
50%         2.000000      0.000000     25.000000      0.637800      0.160100   
75%         3.000000      0.000000     27.000000      0.836750      0.319200   
max         9.000000      8.000000     44.000000      0.995200      0.901200   

              EXPD14  
count   19435.000000  
unique           NaN  
top              NaN  
freq             NaN  
mean        0.094167  
std         0.104473  
min         0.005000  
25%         0.028500  
50%         0.047100  
75%         0.113100  
max         0.576100  

[11 rows x 57 columns]

In [240]:
# Find the number of remaining na values in each column
for col in stroke_trials_df.columns:
    if stroke_trials_df[col].isna().sum() > 0:
        print(f'{col}: {stroke_trials_df[col].isna().sum()}')

RATRIAL: 984
RHEP24: 344
ONDRUG: 1
DOAC: 18
DGORM: 23
DSTER: 28
DCAA: 29
DHAEMD: 28
DMAJNCH: 14
DDIAGISC: 23
DDIAGHA: 26
DDIAGUN: 23
DNOSTRK: 26
DRSISC: 18
DRSH: 15
DPE: 14
DDEAD: 19


In [241]:
# View the rows that have NaN values in RHEP24 column and see how many have 'N', 'U', or NaN values in the other heparin columns (RXHEP, DLH14, DHH14, DMH14, DSCH, DIVH)
rhep24 = stroke_trials_df.loc[stroke_trials_df['RHEP24'].isnull()]
rhep24[['RXHEP','DLH14','DHH14','DMH14','DSCH','DIVH']].value_counts()

RXHEP  DLH14  DHH14  DMH14  DSCH  DIVH
N      N      N      U      U     U       147
L      Y      N      U      U     U        61
H      N      Y      U      U     U        58
N      N      N      U      N     N        23
L      Y      N      U      N     N        13
H      N      N      U      U     U        11
              Y      U      N     N        10
L      N      N      U      U     U         7
H      N      N      U      N     N         5
       Y      N      U      N     N         2
L      N      Y      U      N     N         1
       U      N      U      U     U         1
       N      N      U      N     U         1
                                  N         1
       Y      N      U      Y     N         1
H      Y      N      U      U     U         1
N      N      N      U      Y     Y         1
Name: count, dtype: int64

There are 173 rows that we can use if we save the NaN rows in the 'RHEP24' column (147 + 23 + 1 + 1 + 1)

In [242]:
# Fill na values in RHEP24 column with 'U'
stroke_trials_df['RHEP24'] = stroke_trials_df['RHEP24'].fillna('U')

In [243]:
# Drop all rows with na values
stroke_trials_dropna_df = stroke_trials_df.dropna()
stroke_trials_dropna_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   18405  18405.000000  18405   18405  18405   18405  18405   
unique      2           NaN      2       2      2       2      2   
top         M           NaN      N       N      Y       N      N   
freq     9825           NaN  13011   15243  12372   12382  17977   
mean      NaN     71.802934    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.609051    NaN     NaN    NaN     NaN    NaN   
min       NaN     16.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     98.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ... DPLACE  DDEAD        DDEADC FPLACE  \
count   18405.000000  18405  18405  ...  18405  18405  18405.000000  18405   
unique           NaN      3      3  ...      6      3           NaN      6   
top              NaN      F      Y  ...      U      N           NaN      A   
freq             NaN  14112  13363  ...   9058  16481           NaN  10956   
mean      160.213583    NaN    NaN  ...    NaN    NaN      0.287313    NaN   
std        27.724123    NaN    NaN  ...    NaN    NaN      1.082720    NaN   
min        70.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
25%       140.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
50%       160.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
75%       180.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
max       295.000000    NaN    NaN  ...    NaN    NaN      8.000000    NaN   

              OCCODE        FDEADC      CNTRYNUM         EXPDD         EXPD6  \
count   18405.000000  18405.000000  18405.000000  18405.000000  18405.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        2.327574      0.775061     21.050801      0.631258      0.231408   
std         1.068661      1.830083      8.437599      0.230384      0.195796   
min         0.000000      0.000000      1.000000      0.089100      0.003500   
25%         2.000000      0.000000     14.000000      0.441500      0.085000   
50%         2.000000      0.000000     25.000000      0.641000      0.161300   
75%         3.000000      0.000000     27.000000      0.838900      0.322200   
max         9.000000      8.000000     44.000000      0.995200      0.901200   

              EXPD14  
count   18405.000000  
unique           NaN  
top              NaN  
freq             NaN  
mean        0.095004  
std         0.105377  
min         0.005000  
25%         0.028600  
50%         0.047300  
75%         0.114300  
max         0.576100  

[11 rows x 57 columns]

In [244]:
# Filter out rows that have 'Y' in the RHEP24 column
stroke_trials_rhep24_df = stroke_trials_dropna_df[stroke_trials_dropna_df['RHEP24'] != 'Y']
stroke_trials_rhep24_df = stroke_trials_rhep24_df.reset_index(drop=True)
stroke_trials_rhep24_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   17977  17977.000000  17977   17977  17977   17977  17977   
unique      2           NaN      2       2      2       2      1   
top         M           NaN      N       N      Y       N      N   
freq     9627           NaN  12687   14912  12039   12115  17977   
mean      NaN     71.787006    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.623209    NaN     NaN    NaN     NaN    NaN   
min       NaN     16.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     98.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ... DPLACE  DDEAD        DDEADC FPLACE  \
count   17977.000000  17977  17977  ...  17977  17977  17977.000000  17977   
unique           NaN      3      3  ...      6      3           NaN      6   
top              NaN      F      Y  ...      U      N           NaN      A   
freq             NaN  13784  13034  ...   8858  16095           NaN  10694   
mean      160.310174    NaN    NaN  ...    NaN    NaN      0.287367    NaN   
std        27.749877    NaN    NaN  ...    NaN    NaN      1.082168    NaN   
min        70.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
25%       140.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
50%       160.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
75%       180.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
max       295.000000    NaN    NaN  ...    NaN    NaN      8.000000    NaN   

              OCCODE        FDEADC      CNTRYNUM         EXPDD         EXPD6  \
count   17977.000000  17977.000000  17977.000000  17977.000000  17977.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        2.328865      0.773600     21.154086      0.630842      0.231241   
std         1.070754      1.827973      8.415366      0.230713      0.195795   
min         0.000000      0.000000      1.000000      0.089100      0.003500   
25%         2.000000      0.000000     14.000000      0.440200      0.084900   
50%         2.000000      0.000000     25.000000      0.640400      0.161300   
75%         3.000000      0.000000     27.000000      0.838600      0.321800   
max         9.000000      8.000000     44.000000      0.993500      0.901200   

              EXPD14  
count   17977.000000  
unique           NaN  
top              NaN  
freq             NaN  
mean        0.094947  
std         0.105340  
min         0.005000  
25%         0.028500  
50%         0.047300  
75%         0.114200  
max         0.576100  

[11 rows x 57 columns]

In [245]:
# Filter out rows that have 'Y' in the RXHEP column
stroke_trials_rxhep_df = stroke_trials_rhep24_df[stroke_trials_rhep24_df['RXHEP'] != 'Y']
stroke_trials_rxhep_df = stroke_trials_rxhep_df.drop(columns=['RXHEP'])
stroke_trials_rxhep_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   17977  17977.000000  17977   17977  17977   17977  17977   
unique      2           NaN      2       2      2       2      1   
top         M           NaN      N       N      Y       N      N   
freq     9627           NaN  12687   14912  12039   12115  17977   
mean      NaN     71.787006    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.623209    NaN     NaN    NaN     NaN    NaN   
min       NaN     16.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     98.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ... DPLACE  DDEAD        DDEADC FPLACE  \
count   17977.000000  17977  17977  ...  17977  17977  17977.000000  17977   
unique           NaN      3      3  ...      6      3           NaN      6   
top              NaN      F      Y  ...      U      N           NaN      A   
freq             NaN  13784  13034  ...   8858  16095           NaN  10694   
mean      160.310174    NaN    NaN  ...    NaN    NaN      0.287367    NaN   
std        27.749877    NaN    NaN  ...    NaN    NaN      1.082168    NaN   
min        70.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
25%       140.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
50%       160.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
75%       180.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
max       295.000000    NaN    NaN  ...    NaN    NaN      8.000000    NaN   

              OCCODE        FDEADC      CNTRYNUM         EXPDD         EXPD6  \
count   17977.000000  17977.000000  17977.000000  17977.000000  17977.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        2.328865      0.773600     21.154086      0.630842      0.231241   
std         1.070754      1.827973      8.415366      0.230713      0.195795   
min         0.000000      0.000000      1.000000      0.089100      0.003500   
25%         2.000000      0.000000     14.000000      0.440200      0.084900   
50%         2.000000      0.000000     25.000000      0.640400      0.161300   
75%         3.000000      0.000000     27.000000      0.838600      0.321800   
max         9.000000      8.000000     44.000000      0.993500      0.901200   

              EXPD14  
count   17977.000000  
unique           NaN  
top              NaN  
freq             NaN  
mean        0.094947  
std         0.105340  
min         0.005000  
25%         0.028500  
50%         0.047300  
75%         0.114200  
max         0.576100  

[11 rows x 56 columns]

In [246]:
# Filter out rows that have 'Y' in the DLH14 column
stroke_trials_dlh14_df = stroke_trials_rxhep_df[stroke_trials_rxhep_df['DLH14'] != 'Y']
stroke_trials_dlh14_df = stroke_trials_dlh14_df.drop(columns=['DLH14'])
stroke_trials_dlh14_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   13768  13768.000000  13768   13768  13768   13768  13768   
unique      2           NaN      2       2      2       2      1   
top         M           NaN      N       N      Y       N      N   
freq     7360           NaN   9718   11397   9151    9259  13768   
mean      NaN     71.765035    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.644272    NaN     NaN    NaN     NaN    NaN   
min       NaN     19.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     98.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ... DPLACE  DDEAD        DDEADC FPLACE  \
count   13768.000000  13768  13768  ...  13768  13768  13768.000000  13768   
unique           NaN      3      3  ...      6      3           NaN      6   
top              NaN      F      Y  ...      U      N           NaN      A   
freq             NaN  10552   9974  ...   6799  12305           NaN   8159   
mean      160.071252    NaN    NaN  ...    NaN    NaN      0.291618    NaN   
std        27.720367    NaN    NaN  ...    NaN    NaN      1.088853    NaN   
min        71.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
25%       140.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
50%       160.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
75%       180.000000    NaN    NaN  ...    NaN    NaN      0.000000    NaN   
max       295.000000    NaN    NaN  ...    NaN    NaN      8.000000    NaN   

              OCCODE        FDEADC      CNTRYNUM         EXPDD         EXPD6  \
count   13768.000000  13768.000000  13768.000000  13768.000000  13768.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        2.324303      0.776801     21.202571      0.631075      0.231324   
std         1.068681      1.828388      8.396093      0.230859      0.195794   
min         0.000000      0.000000      1.000000      0.089100      0.004000   
25%         2.000000      0.000000     14.000000      0.441100      0.085000   
50%         2.000000      0.000000     25.000000      0.639750      0.161450   
75%         3.000000      0.000000     27.000000      0.838900      0.322025   
max         9.000000      8.000000     44.000000      0.993500      0.879200   

              EXPD14  
count   13768.000000  
unique           NaN  
top              NaN  
freq             NaN  
mean        0.095039  
std         0.105339  
min         0.005100  
25%         0.028500  
50%         0.047350  
75%         0.114600  
max         0.552200  

[11 rows x 55 columns]

In [247]:
# Filer out rows that have 'Y' in the DMH14 column
stroke_trials_dmh14_df = stroke_trials_dlh14_df[stroke_trials_dlh14_df['DMH14'] != 'Y']
stroke_trials_dmh14_df = stroke_trials_dmh14_df.drop(columns=['DMH14'])
stroke_trials_dmh14_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   9783  9783.000000   9783    9783  9783    9783   9783  9783.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    5203          NaN   6915    8094  6386    6658   9783          NaN   
mean     NaN    71.780435    NaN     NaN   NaN     NaN    NaN   160.170807   
std      NaN    11.709731    NaN     NaN   NaN     NaN    NaN    28.000386   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    80.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ... DPLACE DDEAD       DDEADC FPLACE       OCCODE  \
count    9783  9783  ...   9783  9783  9783.000000   9783  9783.000000   
unique      3     3  ...      6     3          NaN      6          NaN   
top         F     Y  ...      U     N          NaN      A          NaN   
freq     7490  7076  ...   4812  8745          NaN   5751          NaN   
mean      NaN   NaN  ...    NaN   NaN     0.284984    NaN     2.317387   
std       NaN   NaN  ...    NaN   NaN     1.070862    NaN     1.069145   
min       NaN   NaN  ...    NaN   NaN     0.000000    NaN     0.000000   
25%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
50%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
75%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     3.000000   
max       NaN   NaN  ...    NaN   NaN     8.000000    NaN     9.000000   

             FDEADC     CNTRYNUM        EXPDD        EXPD6       EXPD14  
count   9783.000000  9783.000000  9783.000000  9783.000000  9783.000000  
unique          NaN          NaN          NaN          NaN          NaN  
top             NaN          NaN          NaN          NaN          NaN  
freq            NaN          NaN          NaN          NaN          NaN  
mean       0.768169    21.296228     0.633252     0.232819     0.095676  
std        1.814777     8.363065     0.231238     0.196266     0.105560  
min        0.000000     1.000000     0.089100     0.004000     0.005100  
25%        0.000000    14.000000     0.441550     0.085500     0.028600  
50%        0.000000    25.000000     0.645200     0.162700     0.048000  
75%        0.000000    27.000000     0.841550     0.325200     0.116100  
max        8.000000    44.000000     0.993500     0.879200     0.552200  

[11 rows x 54 columns]

In [248]:
# Filter out rows that have 'Y' in the DHH14 column
stroke_trials_dhh14_df = stroke_trials_dmh14_df[stroke_trials_dmh14_df['DHH14'] != 'Y']
stroke_trials_dhh14_df = stroke_trials_dhh14_df.drop(columns=['DHH14'])
stroke_trials_dhh14_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   9783  9783.000000   9783    9783  9783    9783   9783  9783.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    5203          NaN   6915    8094  6386    6658   9783          NaN   
mean     NaN    71.780435    NaN     NaN   NaN     NaN    NaN   160.170807   
std      NaN    11.709731    NaN     NaN   NaN     NaN    NaN    28.000386   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    80.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ... DPLACE DDEAD       DDEADC FPLACE       OCCODE  \
count    9783  9783  ...   9783  9783  9783.000000   9783  9783.000000   
unique      3     3  ...      6     3          NaN      6          NaN   
top         F     Y  ...      U     N          NaN      A          NaN   
freq     7490  7076  ...   4812  8745          NaN   5751          NaN   
mean      NaN   NaN  ...    NaN   NaN     0.284984    NaN     2.317387   
std       NaN   NaN  ...    NaN   NaN     1.070862    NaN     1.069145   
min       NaN   NaN  ...    NaN   NaN     0.000000    NaN     0.000000   
25%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
50%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
75%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     3.000000   
max       NaN   NaN  ...    NaN   NaN     8.000000    NaN     9.000000   

             FDEADC     CNTRYNUM        EXPDD        EXPD6       EXPD14  
count   9783.000000  9783.000000  9783.000000  9783.000000  9783.000000  
unique          NaN          NaN          NaN          NaN          NaN  
top             NaN          NaN          NaN          NaN          NaN  
freq            NaN          NaN          NaN          NaN          NaN  
mean       0.768169    21.296228     0.633252     0.232819     0.095676  
std        1.814777     8.363065     0.231238     0.196266     0.105560  
min        0.000000     1.000000     0.089100     0.004000     0.005100  
25%        0.000000    14.000000     0.441550     0.085500     0.028600  
50%        0.000000    25.000000     0.645200     0.162700     0.048000  
75%        0.000000    27.000000     0.841550     0.325200     0.116100  
max        8.000000    44.000000     0.993500     0.879200     0.552200  

[11 rows x 53 columns]

In [249]:
# Filter out rows that have 'Y' in the DSCH column
stroke_trials_dsch_df = stroke_trials_dhh14_df[stroke_trials_dhh14_df['DSCH'] != 'Y']
stroke_trials_dsch_df = stroke_trials_dsch_df.drop(columns=['DSCH'])
stroke_trials_dsch_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   9446  9446.000000   9446    9446  9446    9446   9446  9446.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    5038          NaN   6669    7823  6139    6433   9446          NaN   
mean     NaN    71.780224    NaN     NaN   NaN     NaN    NaN   160.222316   
std      NaN    11.719854    NaN     NaN   NaN     NaN    NaN    28.011635   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    80.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ... DPLACE DDEAD       DDEADC FPLACE       OCCODE  \
count    9446  9446  ...   9446  9446  9446.000000   9446  9446.000000   
unique      3     3  ...      6     3          NaN      6          NaN   
top         F     Y  ...      U     N          NaN      A          NaN   
freq     7260  6805  ...   4608  8447          NaN   5568          NaN   
mean      NaN   NaN  ...    NaN   NaN     0.279166    NaN     2.322782   
std       NaN   NaN  ...    NaN   NaN     1.054313    NaN     1.070359   
min       NaN   NaN  ...    NaN   NaN     0.000000    NaN     0.000000   
25%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
50%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
75%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     3.000000   
max       NaN   NaN  ...    NaN   NaN     8.000000    NaN     9.000000   

             FDEADC     CNTRYNUM        EXPDD        EXPD6       EXPD14  
count   9446.000000  9446.000000  9446.000000  9446.000000  9446.000000  
unique          NaN          NaN          NaN          NaN          NaN  
top             NaN          NaN          NaN          NaN          NaN  
freq            NaN          NaN          NaN          NaN          NaN  
mean       0.759793    21.362799     0.631587     0.231961     0.095162  
std        1.802121     8.355987     0.231559     0.196209     0.105461  
min        0.000000     1.000000     0.089100     0.004000     0.005100  
25%        0.000000    14.000000     0.439800     0.085000     0.028400  
50%        0.000000    25.000000     0.642250     0.161700     0.047600  
75%        0.000000    27.000000     0.840175     0.322800     0.114775  
max        8.000000    44.000000     0.993500     0.879200     0.552200  

[11 rows x 52 columns]

In [250]:
# Filter out rows that have 'Y' in the DIVH column
stroke_trials_divh_df = stroke_trials_dsch_df[stroke_trials_dsch_df['DIVH'] != 'Y']
stroke_trials_divh_df = stroke_trials_divh_df.drop(columns=['DIVH'])
stroke_trials_divh_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   9198  9198.000000   9198    9198  9198    9198   9198  9198.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    4893          NaN   6486    7620  5938    6282   9198          NaN   
mean     NaN    71.887041    NaN     NaN   NaN     NaN    NaN   160.256143   
std      NaN    11.659757    NaN     NaN   NaN     NaN    NaN    28.077996   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    81.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ... DPLACE DDEAD       DDEADC FPLACE       OCCODE  \
count    9198  9198  ...   9198  9198  9198.000000   9198  9198.000000   
unique      3     3  ...      6     3          NaN      6          NaN   
top         F     Y  ...      U     N          NaN      A          NaN   
freq     7057  6623  ...   4466  8221          NaN   5420          NaN   
mean      NaN   NaN  ...    NaN   NaN     0.279082    NaN     2.323222   
std       NaN   NaN  ...    NaN   NaN     1.052632    NaN     1.071138   
min       NaN   NaN  ...    NaN   NaN     0.000000    NaN     0.000000   
25%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
50%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
75%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     3.000000   
max       NaN   NaN  ...    NaN   NaN     8.000000    NaN     9.000000   

             FDEADC     CNTRYNUM        EXPDD        EXPD6       EXPD14  
count   9198.000000  9198.000000  9198.000000  9198.000000  9198.000000  
unique          NaN          NaN          NaN          NaN          NaN  
top             NaN          NaN          NaN          NaN          NaN  
freq            NaN          NaN          NaN          NaN          NaN  
mean       0.762340    21.373451     0.632587     0.233059     0.095631  
std        1.804109     8.329507     0.231599     0.196902     0.105938  
min        0.000000     1.000000     0.089100     0.004000     0.005100  
25%        0.000000    14.000000     0.440625     0.085400     0.028400  
50%        0.000000    25.000000     0.643100     0.162200     0.047800  
75%        0.000000    27.000000     0.841475     0.325175     0.115600  
max        8.000000    44.000000     0.993500     0.879200     0.552200  

[11 rows x 51 columns]

In [251]:
# Filter out rows that have 'Y' in the DHAEMD column
stroke_trials_dhaemd_df = stroke_trials_divh_df[stroke_trials_divh_df['DHAEMD'] != 'Y']
stroke_trials_dhaemd_df = stroke_trials_dhaemd_df.drop(columns=['DHAEMD'])
stroke_trials_dhaemd_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   8925  8925.000000   8925    8925  8925    8925   8925  8925.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    4750          NaN   6312    7392  5717    6097   8925          NaN   
mean     NaN    71.967619    NaN     NaN   NaN     NaN    NaN   160.238319   
std      NaN    11.641860    NaN     NaN   NaN     NaN    NaN    28.090209   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    81.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ... DPLACE DDEAD       DDEADC FPLACE       OCCODE  \
count    8925  8925  ...   8925  8925  8925.000000   8925  8925.000000   
unique      3     3  ...      6     3          NaN      6          NaN   
top         F     Y  ...      U     N          NaN      A          NaN   
freq     6863  6401  ...   4334  7981          NaN   5250          NaN   
mean      NaN   NaN  ...    NaN   NaN     0.279216    NaN     2.321008   
std       NaN   NaN  ...    NaN   NaN     1.056038    NaN     1.069802   
min       NaN   NaN  ...    NaN   NaN     0.000000    NaN     0.000000   
25%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
50%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
75%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     3.000000   
max       NaN   NaN  ...    NaN   NaN     8.000000    NaN     9.000000   

             FDEADC     CNTRYNUM        EXPDD        EXPD6       EXPD14  
count   8925.000000  8925.000000  8925.000000  8925.000000  8925.000000  
unique          NaN          NaN          NaN          NaN          NaN  
top             NaN          NaN          NaN          NaN          NaN  
freq            NaN          NaN          NaN          NaN          NaN  
mean       0.759440    21.639328     0.632737     0.233161     0.095437  
std        1.799525     8.125983     0.231502     0.196667     0.105806  
min        0.000000     1.000000     0.089100     0.004000     0.005100  
25%        0.000000    15.000000     0.440700     0.085600     0.028400  
50%        0.000000    25.000000     0.644300     0.162900     0.047800  
75%        0.000000    27.000000     0.841600     0.324800     0.115300  
max        8.000000    44.000000     0.993500     0.879200     0.552200  

[11 rows x 50 columns]

In [252]:
# Filter out rows that have 'Y' in the DCAREND column
stroke_trials_dcarend_df = stroke_trials_dhaemd_df[stroke_trials_dhaemd_df['DCAREND'] != 'Y']
stroke_trials_dcarend_df = stroke_trials_dcarend_df.drop(columns=['DCAREND'])
stroke_trials_dcarend_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   8912  8912.000000   8912    8912  8912    8912   8912  8912.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    4741          NaN   6301    7379  5708    6088   8912          NaN   
mean     NaN    71.971836    NaN     NaN   NaN     NaN    NaN   160.224192   
std      NaN    11.640598    NaN     NaN   NaN     NaN    NaN    28.102942   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    81.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ... DPLACE DDEAD       DDEADC FPLACE       OCCODE  \
count    8912  8912  ...   8912  8912  8912.000000   8912  8912.000000   
unique      3     3  ...      6     3          NaN      6          NaN   
top         F     Y  ...      U     N          NaN      A          NaN   
freq     6850  6394  ...   4329  7968          NaN   5240          NaN   
mean      NaN   NaN  ...    NaN   NaN     0.279623    NaN     2.320467   
std       NaN   NaN  ...    NaN   NaN     1.056754    NaN     1.070133   
min       NaN   NaN  ...    NaN   NaN     0.000000    NaN     0.000000   
25%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
50%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
75%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     3.000000   
max       NaN   NaN  ...    NaN   NaN     8.000000    NaN     9.000000   

             FDEADC     CNTRYNUM        EXPDD        EXPD6       EXPD14  
count   8912.000000  8912.000000  8912.000000  8912.000000  8912.000000  
unique          NaN          NaN          NaN          NaN          NaN  
top             NaN          NaN          NaN          NaN          NaN  
freq            NaN          NaN          NaN          NaN          NaN  
mean       0.760548    21.645310     0.633002     0.233344     0.095528  
std        1.800604     8.126827     0.231453     0.196736     0.105854  
min        0.000000     1.000000     0.089100     0.004000     0.005100  
25%        0.000000    15.000000     0.441175     0.085700     0.028500  
50%        0.000000    25.000000     0.644550     0.163000     0.047800  
75%        0.000000    27.000000     0.841725     0.325125     0.115400  
max        8.000000    44.000000     0.993500     0.879200     0.552200  

[11 rows x 49 columns]

In [253]:
# Filter out rows that have 'Y' in the DTHROMB column
stroke_trials_dthromb_df = stroke_trials_dcarend_df[stroke_trials_dcarend_df['DTHROMB'] != 'Y']
stroke_trials_remove_hep_df = stroke_trials_dthromb_df.drop(columns=['DTHROMB'])
stroke_trials_remove_hep_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   8907  8907.000000   8907    8907  8907    8907   8907  8907.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    4740          NaN   6297    7377  5704    6084   8907          NaN   
mean     NaN    71.970809    NaN     NaN   NaN     NaN    NaN   160.227686   
std      NaN    11.642180    NaN     NaN   NaN     NaN    NaN    28.108205   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    81.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ... DPLACE DDEAD      DDEADC FPLACE       OCCODE  \
count    8907  8907  ...   8907  8907  8907.00000   8907  8907.000000   
unique      3     3  ...      6     3         NaN      6          NaN   
top         F     Y  ...      U     N         NaN      A          NaN   
freq     6846  6390  ...   4328  7963         NaN   5235          NaN   
mean      NaN   NaN  ...    NaN   NaN     0.27978    NaN     2.320310   
std       NaN   NaN  ...    NaN   NaN     1.05703    NaN     1.070245   
min       NaN   NaN  ...    NaN   NaN     0.00000    NaN     0.000000   
25%       NaN   NaN  ...    NaN   NaN     0.00000    NaN     2.000000   
50%       NaN   NaN  ...    NaN   NaN     0.00000    NaN     2.000000   
75%       NaN   NaN  ...    NaN   NaN     0.00000    NaN     3.000000   
max       NaN   NaN  ...    NaN   NaN     8.00000    NaN     9.000000   

             FDEADC     CNTRYNUM        EXPDD        EXPD6      EXPD14  
count   8907.000000  8907.000000  8907.000000  8907.000000  8907.00000  
unique          NaN          NaN          NaN          NaN         NaN  
top             NaN          NaN          NaN          NaN         NaN  
freq            NaN          NaN          NaN          NaN         NaN  
mean       0.760975    21.647693     0.632961     0.233323     0.09552  
std        1.801019     8.126888     0.231457     0.196755     0.10586  
min        0.000000     1.000000     0.089100     0.004000     0.00510  
25%        0.000000    15.000000     0.441150     0.085700     0.02850  
50%        0.000000    25.000000     0.644400     0.162900     0.04780  
75%        0.000000    27.000000     0.841650     0.325100     0.11540  
max        8.000000    44.000000     0.993500     0.879200     0.55220  

[11 rows x 48 columns]

In [254]:
# Create dataFrames of target variables
target_14_days_df = stroke_trials_remove_hep_df[['DALIVE', 'DPLACE', 'DDEAD', 'DDEADC']].copy()
target_6_months_df = stroke_trials_remove_hep_df[['FPLACE', 'OCCODE', 'FDEADC']].copy()
target_14_days_df.head()

DALIVE DPLACE DDEAD  DDEADC
1      N      U     Y     4.0
2      Y      A     N     0.0
4      N      U     N     0.0
5      N      U     Y     4.0
6      Y      A     N     0.0

In [255]:
# See the value counts of each column in the target_14_days_df
for col in target_14_days_df.columns:
    print(f"{target_14_days_df[col].value_counts()}")

DALIVE
Y    4791
N    4105
U      11
Name: count, dtype: int64
DPLACE
U    4328
A    3325
E     735
D     261
C     137
B     121
Name: count, dtype: int64
DDEAD
N    7963
Y     938
U       6
Name: count, dtype: int64
DDEADC
0.0    7969
1.0     472
4.0     167
2.0     106
5.0      71
6.0      41
7.0      39
8.0      28
3.0      14
Name: count, dtype: int64


- Drop rows with 'U' in DALIVE and DDEAD columns (only small number of rows)
- Decide what to do with DPLACE and DDEADC columns (drop DPLACE column, drop DDEADC column, drop 'U' rows in DPLACE, drop '0' rows in DDEADC, or a combination of those) - Going to drop both columns for now. 

In [256]:
# View the DPLACE, DALIVE, and DDEADC value counts for rows that have 'U' in the DDEAD column
ddead = stroke_trials_remove_hep_df[stroke_trials_remove_hep_df['DDEAD'] == 'U']
ddead[['DPLACE','DALIVE','DDEADC']].value_counts()

DPLACE  DALIVE  DDEADC
U       U       0.0       6
Name: count, dtype: int64

In [257]:
# Drop rows that have 'U' in the DALIVE and DDEAD columns
stroke_trials_remove_hep_df = stroke_trials_remove_hep_df[(stroke_trials_remove_hep_df['DALIVE'] != 'U') & (stroke_trials_remove_hep_df['DDEAD'] != 'U')]
stroke_trials_remove_hep_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   8896  8896.000000   8896    8896  8896    8896   8896  8896.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    4733          NaN   6289    7368  5699    6076   8896          NaN   
mean     NaN    71.971111    NaN     NaN   NaN     NaN    NaN   160.224483   
std      NaN    11.644160    NaN     NaN   NaN     NaN    NaN    28.103412   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    81.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ... DPLACE DDEAD       DDEADC FPLACE       OCCODE  \
count    8896  8896  ...   8896  8896  8896.000000   8896  8896.000000   
unique      3     3  ...      6     2          NaN      6          NaN   
top         F     Y  ...      U     N          NaN      A          NaN   
freq     6839  6384  ...   4317  7963          NaN   5232          NaN   
mean      NaN   NaN  ...    NaN   NaN     0.279227    NaN     2.319469   
std       NaN   NaN  ...    NaN   NaN     1.056812    NaN     1.065693   
min       NaN   NaN  ...    NaN   NaN     0.000000    NaN     0.000000   
25%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
50%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     2.000000   
75%       NaN   NaN  ...    NaN   NaN     0.000000    NaN     3.000000   
max       NaN   NaN  ...    NaN   NaN     8.000000    NaN     9.000000   

             FDEADC     CNTRYNUM        EXPDD        EXPD6       EXPD14  
count   8896.000000  8896.000000  8896.000000  8896.000000  8896.000000  
unique          NaN          NaN          NaN          NaN          NaN  
top             NaN          NaN          NaN          NaN          NaN  
freq            NaN          NaN          NaN          NaN          NaN  
mean       0.761016    21.639501     0.632939     0.233244     0.095459  
std        1.801689     8.127821     0.231377     0.196667     0.105771  
min        0.000000     1.000000     0.089100     0.004000     0.005100  
25%        0.000000    15.000000     0.441275     0.085700     0.028500  
50%        0.000000    25.000000     0.644350     0.162900     0.047800  
75%        0.000000    27.000000     0.841600     0.324800     0.115400  
max        8.000000    44.000000     0.993500     0.879200     0.552200  

[11 rows x 48 columns]

In [258]:
# Drop DPLACE and DDEADC
stroke_trials_remove_hep_df = stroke_trials_remove_hep_df.drop(columns=['DPLACE', 'DDEADC'])

In [259]:
# See the value counts of each column in the target_6_months_df
for col in target_6_months_df.columns:
    print(f"{target_6_months_df[col].value_counts()}")

FPLACE
A    5235
U    2109
D     618
B     402
C     337
E     206
Name: count, dtype: int64
OCCODE
2    3636
1    1954
3    1774
4    1469
0      47
9      27
Name: count, dtype: int64
FDEADC
0.0    6892
1.0     737
4.0     394
2.0     256
7.0     219
5.0     179
8.0     119
6.0      80
3.0      31
Name: count, dtype: int64


- Drop rows with '0' or '9' in OCCODE column (only small number of rows)
- Decide what to do with FPLACE and FDEADC columns (drop FPLACE column, drop FDEADC column, drop rows with 'U' in FPLACE, drop rows with '0' in FDEADC, or combination of those) - Going to drop both columns for now.

In [260]:
# View the FPLACE and OCCODE value counts for rows that have 0 in the FDEADC column
fdeadc = stroke_trials_remove_hep_df[stroke_trials_remove_hep_df['FDEADC'] == 0]
fdeadc[['FPLACE','OCCODE']].value_counts()

FPLACE  OCCODE
A       2         2250
        3         1627
        4         1355
D       2          601
B       2          262
C       2          250
E       2          192
B       3           85
        4           55
C       3           47
U       0           47
C       4           40
U       9           25
        2           13
D       4           10
E       4            8
D       3            7
E       3            6
U       1            5
        4            1
Name: count, dtype: int64

In [261]:
# Drop rows with '0' in the OCCODE column
stroke_trials_remove_hep_df = stroke_trials_remove_hep_df[(stroke_trials_remove_hep_df['OCCODE'] != 0) & (stroke_trials_remove_hep_df['OCCODE'] != 9)]
stroke_trials_remove_hep_df['OCCODE'].nunique()

4

In [262]:
# Drop FPLACE and FDEADC columns
stroke_trials_remove_hep_df = stroke_trials_remove_hep_df.drop(columns=['FPLACE', 'FDEADC'])

In [263]:
stroke_trials_remove_hep_df.columns

Index(['SEX', 'AGE', 'RSLEEP', 'RATRIAL', 'RCT', 'RVISINF', 'RHEP24', 'RSBP',
       'RCONSC', 'RDEF1', 'RDEF2', 'RDEF3', 'RDEF4', 'RDEF5', 'RDEF6', 'RDEF7',
       'RDEF8', 'STYPE', 'RDATE', 'RXASP', 'DASP14', 'DASPLT', 'ONDRUG', 'DAP',
       'DOAC', 'DGORM', 'DSTER', 'DCAA', 'DMAJNCH', 'DDIAGISC', 'DDIAGHA',
       'DDIAGUN', 'DNOSTRK', 'DRSISC', 'DRSH', 'DRSUNK', 'DPE', 'DALIVE',
       'DDEAD', 'OCCODE', 'CNTRYNUM', 'EXPDD', 'EXPD6', 'EXPD14'],
      dtype='object')

Potential Later Options:
- Add DALIVE to six_month_df
- Create fourteen_day_df with DDEAD, DALIVE, DDEADC, or DPLACE as the y
- Replace OCCODE with FPLACE or FDEADC as the y

In [264]:
# Get important columns for the model
six_month_df = stroke_trials_remove_hep_df[['SEX', 'AGE', 'RSLEEP', 'RATRIAL', 'RCT', 'RVISINF', 'RHEP24', 'RSBP',
       'RCONSC', 'RDEF1', 'RDEF2', 'RDEF3', 'RDEF4', 'RDEF5', 'RDEF6', 'RDEF7',
       'RDEF8', 'STYPE', 'RDATE', 'RXASP', 'DASP14', 'DASPLT', 'ONDRUG', 'DAP',
       'DOAC', 'DGORM', 'DSTER', 'DCAA', 'DMAJNCH', 'DDIAGISC', 'DDIAGHA',
       'DDIAGUN', 'DNOSTRK', 'DRSISC', 'DRSH', 'DRSUNK', 'DPE', 'CNTRYNUM', 'OCCODE']]
six_month_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8824 entries, 1 to 17975
Data columns (total 39 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   SEX       8824 non-null   object 
 1   AGE       8824 non-null   int64  
 2   RSLEEP    8824 non-null   object 
 3   RATRIAL   8824 non-null   object 
 4   RCT       8824 non-null   object 
 5   RVISINF   8824 non-null   object 
 6   RHEP24    8824 non-null   object 
 7   RSBP      8824 non-null   int64  
 8   RCONSC    8824 non-null   object 
 9   RDEF1     8824 non-null   object 
 10  RDEF2     8824 non-null   object 
 11  RDEF3     8824 non-null   object 
 12  RDEF4     8824 non-null   object 
 13  RDEF5     8824 non-null   object 
 14  RDEF6     8824 non-null   object 
 15  RDEF7     8824 non-null   object 
 16  RDEF8     8824 non-null   object 
 17  STYPE     8824 non-null   object 
 18  RDATE     8824 non-null   object 
 19  RXASP     8824 non-null   object 
 20  DASP14    8824 non-null   object 


In [265]:
six_month_df.head()

SEX  AGE RSLEEP RATRIAL RCT RVISINF RHEP24  RSBP RCONSC RDEF1  ... DDIAGISC  \
1   F   73      Y       N   Y       N      N   120      F     Y  ...        Y   
2   M   74      N       N   Y       Y      N   160      F     Y  ...        Y   
4   M   54      N       N   Y       N      N   160      F     Y  ...        Y   
5   F   79      N       N   N       N      N   175      D     Y  ...        N   
6   M   80      N       Y   Y       Y      N   200      F     Y  ...        Y   

  DDIAGHA DDIAGUN DNOSTRK DRSISC DRSH DRSUNK DPE CNTRYNUM OCCODE  
1       N       N       N      N    N      N   N       27      1  
2       N       N       N      N    N      N   N       14      2  
4       N       N       N      N    N      N   N       27      2  
5       Y       N       N      N    N      N   N       27      1  
6       N       N       N      N    N      N   N       14      2  

[5 rows x 39 columns]

In [266]:
# Filter object columns
columns_to_encode = six_month_df.columns[six_month_df.dtypes == 'object']
columns_to_encode

Index(['SEX', 'RSLEEP', 'RATRIAL', 'RCT', 'RVISINF', 'RHEP24', 'RCONSC',
       'RDEF1', 'RDEF2', 'RDEF3', 'RDEF4', 'RDEF5', 'RDEF6', 'RDEF7', 'RDEF8',
       'STYPE', 'RDATE', 'RXASP', 'DASP14', 'DASPLT', 'DAP', 'DOAC', 'DGORM',
       'DSTER', 'DCAA', 'DMAJNCH', 'DDIAGISC', 'DDIAGHA', 'DDIAGUN', 'DNOSTRK',
       'DRSISC', 'DRSH', 'DRSUNK', 'DPE'],
      dtype='object')

In [267]:
# View sample data in columns to encode
six_month_df[['RDEF2', 'RDEF3', 'RDEF4', 'RDEF5', 'RDEF6', 'RDEF7', 'RDEF8',
       'STYPE', 'RDATE', 'RXASP', 'DASP14', 'DASPLT', 'DAP', 'DOAC', 'DGORM',
       'DSTER', 'DCAA']].head()

RDEF2 RDEF3 RDEF4 RDEF5 RDEF6 RDEF7 RDEF8 STYPE    RDATE RXASP DASP14  \
1     Y     Y     N     N     N     N     N  LACS  1993-03     N      N   
2     Y     N     Y     N     N     N     N  PACS  1993-03     N      Y   
4     N     N     N     N     N     Y     N  POCS  1993-03     Y      Y   
5     Y     Y     Y     Y     Y     N     N  TACS  1993-03     N      N   
6     Y     Y     Y     N     N     N     N  PACS  1993-03     N      N   

  DASPLT DAP DOAC DGORM DSTER DCAA  
1      N   N    N     N     N    N  
2      Y   Y    N     N     N    N  
4      Y   N    N     N     N    Y  
5      N   N    N     N     N    N  
6      Y   N    N     Y     N    Y

In [268]:
#Transform the RDATE column to datetime
pd.options.mode.copy_on_write = True
six_month_df['RDATE'] = pd.to_datetime(six_month_df['RDATE'], format='%Y-%m', errors='coerce')
six_month_df['RDATE'].value_counts(dropna=False)

RDATE
1995-11-01    320
1995-10-01    311
1995-12-01    299
1996-02-01    276
1996-01-01    274
1996-03-01    263
1995-03-01    261
1994-12-01    260
1994-03-01    258
1995-05-01    256
1994-11-01    254
1995-04-01    253
1995-08-01    251
1995-09-01    251
1994-05-01    249
1994-10-01    246
1995-01-01    242
1994-06-01    241
1994-04-01    238
1994-09-01    238
1995-07-01    232
1995-02-01    231
1994-01-01    230
1994-07-01    228
1994-02-01    226
1994-08-01    226
1996-05-01    219
1996-04-01    218
1995-06-01    211
1993-11-01    206
1993-12-01    203
1993-10-01    182
1993-09-01    169
1993-08-01    160
1993-07-01    146
1993-06-01    144
1993-05-01    142
1993-04-01    123
1993-03-01     87
Name: count, dtype: int64

In [269]:
# Update columns_to_encode variable
#columns_to_encode = six_month_df.columns[six_month_df.dtypes == 'object']
columns_to_encode

Index(['SEX', 'RSLEEP', 'RATRIAL', 'RCT', 'RVISINF', 'RHEP24', 'RCONSC',
       'RDEF1', 'RDEF2', 'RDEF3', 'RDEF4', 'RDEF5', 'RDEF6', 'RDEF7', 'RDEF8',
       'STYPE', 'RDATE', 'RXASP', 'DASP14', 'DASPLT', 'DAP', 'DOAC', 'DGORM',
       'DSTER', 'DCAA', 'DMAJNCH', 'DDIAGISC', 'DDIAGHA', 'DDIAGUN', 'DNOSTRK',
       'DRSISC', 'DRSH', 'DRSUNK', 'DPE'],
      dtype='object')

In [270]:
# Create an instance of the label encoder
le = LabelEncoder()

# Fit and transform the label encoder for each column
for column in columns_to_encode:  
    six_month_df[column] = le.fit_transform(six_month_df[column])

six_month_df.head()

SEX  AGE  RSLEEP  RATRIAL  RCT  RVISINF  RHEP24  RSBP  RCONSC  RDEF1  ...  \
1    0   73       1        0    1        0       0   120       1      2  ...   
2    1   74       0        0    1        1       0   160       1      2  ...   
4    1   54       0        0    1        0       0   160       1      2  ...   
5    0   79       0        0    0        0       0   175       0      2  ...   
6    1   80       0        1    1        1       0   200       1      2  ...   

   DDIAGISC  DDIAGHA  DDIAGUN  DNOSTRK  DRSISC  DRSH  DRSUNK  DPE  CNTRYNUM  \
1         1        0        0        0       0     0       0    0        27   
2         1        0        0        0       0     0       0    0        14   
4         1        0        0        0       0     0       0    0        27   
5         0        1        0        0       0     0       0    0        27   
6         1        0        0        0       0     0       0    0        14   

   OCCODE  
1       1  
2       2  
4       2  
5       1  
6       2  

[5 rows x 39 columns]

In [271]:
# Check if there are any columns left to encode
six_month_df.columns[six_month_df.dtypes == 'object']

Index([], dtype='object')

In [272]:
six_month_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8824 entries, 1 to 17975
Data columns (total 39 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   SEX       8824 non-null   int32  
 1   AGE       8824 non-null   int64  
 2   RSLEEP    8824 non-null   int32  
 3   RATRIAL   8824 non-null   int32  
 4   RCT       8824 non-null   int32  
 5   RVISINF   8824 non-null   int32  
 6   RHEP24    8824 non-null   int32  
 7   RSBP      8824 non-null   int64  
 8   RCONSC    8824 non-null   int32  
 9   RDEF1     8824 non-null   int32  
 10  RDEF2     8824 non-null   int32  
 11  RDEF3     8824 non-null   int32  
 12  RDEF4     8824 non-null   int32  
 13  RDEF5     8824 non-null   int32  
 14  RDEF6     8824 non-null   int32  
 15  RDEF7     8824 non-null   int32  
 16  RDEF8     8824 non-null   int32  
 17  STYPE     8824 non-null   int32  
 18  RDATE     8824 non-null   int64  
 19  RXASP     8824 non-null   int32  
 20  DASP14    8824 non-null   int32  


In [273]:
# Create csv files for the six month data
six_month_df.to_csv('Resources/six_month_data.csv', index=False)

In [274]:
# Create dataFrame of doctor's predicted probabilities
doctor_pred_probs_df = stroke_trials_remove_hep_df[['EXPDD','EXPD6']].copy()
doctor_pred_probs_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8824 entries, 1 to 17975
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   EXPDD   8824 non-null   float64
 1   EXPD6   8824 non-null   float64
dtypes: float64(2)
memory usage: 206.8 KB


In [275]:
# Create csv file for doctor's predicted probabilities
doctor_pred_probs_df.to_csv('Resources/doctor_6month_pred.csv', index=False)